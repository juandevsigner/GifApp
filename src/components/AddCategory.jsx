import { useState } from "react";
import Proptypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = e => {
    e.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form aria-label="form" onSubmit={onSubmit}>
      <input
        type="text"
        value={inputValue}
        onChange={onInputChange}
        placeholder="Search Gifs"
      />
    </form>
  );
};

AddCategory.proptypes = {
  onNewCategory: Proptypes.func.isRequired,
};
