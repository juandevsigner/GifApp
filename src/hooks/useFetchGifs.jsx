import { useEffect, useState } from "react";
import { getGift } from "../helpers/getGifs";

export const useFetchGifs = category => {
  const [images, setImages] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    const getImages = async () => {
      const newImages = await getGift(category);
      setImages(newImages);
      setIsLoading(false);
    };
    getImages();
  }, []);

  return { images, isLoading };
};
