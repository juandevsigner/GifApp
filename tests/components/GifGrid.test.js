import { render, screen } from "@testing-library/react"
import { GifGrid } from "../../src/components/GifGrid"
import { useFetchGifs } from "../../src/hooks/useFetchGifs"
jest.mock("../../src/hooks/useFetchGifs")

describe('GifGrid text', () =>{
    useFetchGifs.mockReturnValue({
    images: [],
    isLoading: true,
  });
    const category = 'One Punch'
    it('vs sanpshot', () =>{
       const {container} = render(<GifGrid category={category} />)
        expect(container).toMatchSnapshot()
    })
    it('should show loading text and text category first time', () =>{
         render(<GifGrid category={category} />)
         expect(screen.getByText('One Punch')).toBeTruthy()

    })
    it('should show images', () =>{
        const gifs = [{
            id: '1',
            title: 'One Punch',
            url: "www.imagen.com"
        },{
            id: '2',
            title: 'Goku',
            url: "www.imagen.com"
        }]

        useFetchGifs.mockReturnValue({
            images: gifs,
            isLoading: false,
        });
        render(<GifGrid category={category} />)
        expect(screen.getByText('Goku')).toBeTruthy()
        expect(screen.getAllByRole('img').length).toBe(2)
    })
})