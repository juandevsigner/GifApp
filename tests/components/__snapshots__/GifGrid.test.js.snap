// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid text vs sanpshot 1`] = `
<div>
  <h3>
    One Punch
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
